packages/paste-core/components/chat-composer/__tests__/index.spec.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react';
+import {screen, render, waitFor, fireEvent} from '@testing-library/react';
+import {$getRoot, $createParagraphNode, $createTextNode} from '@twilio-paste/lexical-library';
+import {Theme} from '@twilio-paste/theme';
+
+import {ChatComposer} from '../src';
+
+const initialText = (): void => {
+  const root = $getRoot();
+
+  if (root.getFirstChild() === null) {
+    const paragraph = $createParagraphNode();
+    paragraph.append($createTextNode('Hello'));
+
+    root.append(paragraph);
+  }
+};
+
+const baseConfig = {
+  namespace: 'foo',
+  onError: (error: Error) => {
+    throw error;
+  },
+};
+
+describe('ChatComposer', () => {
+  it('should render with placeholder text', () => {
+    render(<ChatComposer testid="my-composer" placeholder="Type here.." config={baseConfig} />);
+    expect(screen.getByRole('textbox')).toBeDefined();
+    expect(screen.getByText('Type here..')).toBeDefined();
+  });
+
+  it('should pass props to the content editable', async () => {
+    render(
+      <ChatComposer
+        testid="my-composer"
+        ariaLabel="Feedback"
+        ariaDescribedBy="foo"
+        ariaOwns="foo"
+        ariaActiveDescendant="foo"
+        config={baseConfig}
+      />
+    );
+
+    const contentEditable = screen.getByRole('textbox');
+    await waitFor(() => {
+      expect(contentEditable).toHaveAttribute('aria-label', 'Feedback');
+      expect(contentEditable).toHaveAttribute('aria-activedescendant', 'foo');
+      expect(contentEditable).toHaveAttribute('aria-owns', 'foo');
+      expect(contentEditable).toHaveAttribute('aria-describedby', 'foo');
+      expect(contentEditable.dataset.testid).toEqual('my-composer');
+    });
+  });
+
+  it('should render initial value with the initialValue prop', async () => {
+    render(<ChatComposer initialValue="Type here.." config={baseConfig} />);
+
+    const contentEditable = screen.getByRole('textbox');
+    await waitFor(() => {
+      expect(contentEditable).toHaveTextContent('Type here..');
+    });
+  });
+
+  it('should render custom initial value with the config prop', async () => {
+    render(
+      <ChatComposer
+        config={{
+          ...baseConfig,
+          editorState: initialText,
+        }}
+      />
+    );
+
+    const contentEditable = screen.getByRole('textbox');
+    await waitFor(() => {
+      expect(contentEditable).toHaveTextContent('Hello');
+    });
+  });
+
+  it('should set maxHeight with the maxHeight prop', async () => {
+    render(
+      <Theme.Provider theme="default">
+        <ChatComposer initialValue="Type here.." maxHeight="size10" config={baseConfig} />
+      </Theme.Provider>
+    );
+
+    const wrapper = screen.getByRole('textbox').parentElement;
+    await waitFor(() => {
+      expect(wrapper).toHaveStyleRule('max-height', '5.5rem');
+    });
+  });
+
+  /*
```

Comment:
OPEN QUESTION: I am having trouble getting the onChange test to work. I have a story with an onChange that is working as expected, but not sure what the issue is here. Would maybe like to pair with someone on this.

Reply:
I'm down to pair or at least watch the pairing magic happen

Reply:
I figured it out! I was using `fireEvent`, which uses the browser events.. I swapped it to `userEvent` and that did work.

I'm guessing its because lexical doesn't directly use the normal onChange browser event - so `userEvent` worked because it more closely simulates what people do?